src/Donations/Endpoints/DonationUpdate.php

```diff
@@ -0,0 +1,144 @@
+<?php
+
+namespace Give\Donations\Endpoints;
+
+use Give\Donations\Endpoints\DonationUpdateAttributes\AmountAttribute;
+use Give\Donations\Endpoints\DonationUpdateAttributes\AttributeUpdatesModel;
+use Give\Donations\Endpoints\DonationUpdateAttributes\CreatedAtAttribute;
+use Give\Donations\Endpoints\DonationUpdateAttributes\FeeAmountRecoveredAttribute;
+use Give\Donations\Endpoints\DonationUpdateAttributes\FormIdAttribute;
+use Give\Donations\Endpoints\DonationUpdateAttributes\IdAttribute;
+use Give\Donations\Endpoints\DonationUpdateAttributes\StatusAttribute;
+use Give\Framework\Exceptions\Primitives\Exception;
+use WP_Error;
+use WP_REST_Request;
+use WP_REST_Response;
+use WP_REST_Server;
+
+/*
+ * Class DonationDetails
+ *
+ * @unreleased
+ */
+
+class DonationUpdate extends Endpoint
+{
+    /**
+     * @var string
+     */
+    protected $endpoint = 'admin/donation/(?P<id>[\d]+)';
+
+    /**
+     * @var array
+     */
+    private $attributes = [
+        IdAttribute::class,
+        StatusAttribute::class,
+        AmountAttribute::class,
+        FeeAmountRecoveredAttribute::class,
+        FormIdAttribute::class,
+        CreatedAtAttribute::class,
+    ];
+
+    /**
+     * @unreleased
+     *
+     * @inheritDoc
+     */
+    public function registerRoute()
+    {
+        register_rest_route(
+            'give-api/v2',
+            $this->endpoint,
+            [
+                [
+                    'methods' => WP_REST_Server::EDITABLE,
+                    'callback' => [$this, 'handleRequest'],
+                    'permission_callback' => [$this, 'permissionsCheck'],
+                ],
+                'args' => $this->getRouteAttributes(),
+            ]
+        );
+    }
+
+    /**
+     * @unreleased
+     *
+     * @inheritDoc
+     */
+    public function permissionsCheck()
+    {
+        if ( ! current_user_can('edit_give_payments')) {
+            return new WP_Error(
+                'rest_forbidden',
+                esc_html__('You don\'t have permission to edit Donations', 'give'),
+                ['status' => $this->authorizationStatusCode()]
+            );
+        }
+
+        return true;
+    }
+
+    /**
+     * @unreleased
+     *
+     * @param WP_REST_Request $request
+     *
+     * @return WP_REST_Response|WP_Error
+     */
+    public function handleRequest(WP_REST_Request $request)
+    {
+        $id = $request->get_param('id');
+        $donation = give()->donations->getById($id);
+
+        $updatedFields = [];
+
+        try {
+            foreach ($this->attributes as $attr) {
+                $attrId = $attr::getId();
+                $value = $request->get_param($attrId);
+
+                if (is_null($value) || ! in_array(AttributeUpdatesModel::class, class_implements($attr))) {
```

Comment:
@pauloiankoski This reads strangely and accesses more than is necessary. I suggest simplifying with an operator:
```php
if ( $value === null || !$attr instanceof AttributeUpdatesModel::class ) {
```

Also, would it make sense to allow the attribute to determine what it does will a null value? Wouldn't the attribute's validation catch and prevent values it won't accept? If so, the null check isn't necessary.

Reply:
About the operator, that way you propose only works in case `$attr` was an object. In this case, it's just a static class.
![CleanShot 2023-03-29 at 21 02 00](https://user-images.githubusercontent.com/3921017/228694624-7b45be08-2771-4767-9b1d-2f53dd89b67d.jpg)

![CleanShot 2023-03-29 at 21 01 39](https://user-images.githubusercontent.com/3921017/228694580-53e1d165-653b-4349-8a22-a823e1872cf9.jpg)

--------------

Here, I'm looping through all registered attributes then checking if the user passed a value to it then maybe skip or update it. The validation only prevents the user to send invalid values.

![CleanShot 2023-03-29 at 20 56 29](https://user-images.githubusercontent.com/3921017/228693990-b324239e-0ead-48f6-8121-f98c4a244955.jpg)

Reply:
Ah, in that case you can use [is_a](https://www.php.net/manual/en/function.is-a.php): `is_a($attr, AttributeUpdatesModel::class, true)`

Reply:
That worked! 47290827573fa912eed765f98ea7c1626ab80a65

Reply:
> Here, I'm looping through all registered attributes then checking if the user passed a value to it then maybe skip or update it. The validation only prevents the user to send invalid values.

Ah, so this is intended to allow the front-end to send partial data? As in your screenshot, it would update the status and nothing else?

Reply:
Yeah. That's correct. I just have noticed that the front-end is not passing only the dirty fields but all of them. As I created this route before knowing that, I made it in a way that it can receive partial data.